### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by winny-**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM ruby:2.5
+
+# throw errors if Gemfile has been modified since Gemfile.lock
+RUN bundle config --global frozen 1
```
Comment (woodruffw) [inline Dockerfile]: It makes sense to also run `bundle` from within the container, no?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+WORKDIR /usr/src/app
+
+COPY Gemfile Gemfile.lock ./
+
+# https://github.com/jtoy/cld/issues/10
+RUN env CFLAGS=-Wno-narrowing CXXFLAGS=-Wno-narrowing bundle install
```
Comment (woodruffw) [inline Dockerfile]: Does this actually have an effect? I don't see where `cld` is compiled below, but maybe I'm missing something.

Comment (winny-) [reply Dockerfile]: `cld (0.7.0)` is needed for `megahal`. Alternately maybe one can bump `megahal` to depend on a newer version of `cld` where it has been fixed. To test: `docker build -t yossarian-bot:latest` with the two env variables unset. [Here is a build log](https://gist.github.com/winny-/32e7ddf690ea2c5cc5c0e72e8f8ffa0a).

My main motivation to Dockerize is I can't seem to get nokogumbo to compile without doing something awful like `sudo gem install ...`. Thankfully it just works in the container. [This sounds like the problem I'm encountering](https://github.com/rubys/nokogumbo/issues/40). Magic!

File: Dockerfile (hunk 1 above)
Comment (winny-) [reply Dockerfile]: `RUN` runs things in the container before it is made into an image (the end result of a Dockerfile).

File: Dockerfile (hunk 2 above)
Comment (winny-) [reply Dockerfile]: Fwiw I've encountered this problem with cld 0.7.0 on Gentoo and Archlinux. Reportedly it occurs on some versions of Ubuntu too.

**Comment by woodruffw** [comment]
Thanks!

**Pull request merged**